Sources/Swift/Integrations/UserFeedback/SentryUserFeedbackFormController.swift

```diff
@@ -0,0 +1,204 @@
+    lazy var viewModel = SentryUserFeedbackFormViewModel(config: config, controller: self)
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        config.theme.updateDefaultFonts()
+        config.recalculateScaleFactors()
+        viewModel.updateLayout()
+    }
+    
+    init(config: SentryUserFeedbackConfiguration, delegate: any SentryUserFeedbackFormDelegate) {
+        self.config = config
+        self.delegate = delegate
+        super.init(nibName: nil, bundle: nil)
+        view.backgroundColor = config.theme.background
+        initLayout()
+        viewModel.themeElements()
+        
+        let nc = NotificationCenter.default
+        nc.addObserver(self, selector: #selector(showedKeyboard(note:)), name: UIResponder.keyboardDidShowNotification, object: nil)
+        nc.addObserver(self, selector: #selector(hidKeyboard), name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: Layout
+@available(iOS 13.0, *)
+extension SentryUserFeedbackFormController {
+    func initLayout() {
+        viewModel.setScrollViewBottomInset(0)
+        view.addSubview(viewModel.scrollView)
+        NSLayoutConstraint.activate(viewModel.allConstraints(view: view))
+    }
+    
+    func showedKeyboard(note: Notification) {
+        guard let keyboardValue = note.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue else {
+            SentryLog.warning("Received a keyboard display notification with no frame information.")
+            return
+        }
+        let keyboardViewEndFrame = self.view.convert(keyboardValue.cgRectValue, from: self.view.window)
+        viewModel.setScrollViewBottomInset(keyboardViewEndFrame.height - self.view.safeAreaInsets.bottom)
+    }
+    
+    func hidKeyboard() {
+        viewModel.setScrollViewBottomInset(0)
+    }
+}
+
+// MARK: SentryUserFeedbackFormViewModelDelegate
+@available(iOS 13.0, *)
+extension SentryUserFeedbackFormController: SentryUserFeedbackFormViewModelDelegate {
+    public func addScreenshotTapped() {
+        
+#if SENTRY_TEST || SENTRY_TEST_CI || TEST || TESTCI
+        // the iOS photo picker UI doesn't play nicely with XCUITest, so we need to mock it. we also mock it for unit tests
+        set(image: UIImage(), accessibilityInfo: "test image accessibility info")
+#else
+        func presentPicker() {
+            let imagePickerController = UIImagePickerController()
+            imagePickerController.delegate = self
+            imagePickerController.sourceType = .photoLibrary
+            imagePickerController.allowsEditing = true
+            DispatchQueue.main.async {
+                self.present(imagePickerController, animated: self.config.animations)
+            }
+        }
+        
+        let status = PHPhotoLibrary.authorizationStatus()
```

Comment:
`m`: IIRC we discussed that requesting access without a usage description in the `Info.plist` can cause app crashes. How did you solve this?

Reply:
Yes, this is handled in https://github.com/getsentry/sentry-cocoa/pull/4723

Reply:
Yes, this will be further refined in https://github.com/getsentry/sentry-cocoa/pull/4723